src/catch2/catch_tostring.hpp

```diff
@@ -41,6 +41,13 @@ namespace Catch {
 
     namespace Detail {
 
+        inline std::size_t catch_strnlen(const char *str, std::size_t n) {
+            auto ret = std::memchr(str, '\0', n);
+            if (ret != nullptr) {
+                return static_cast<const char *>(ret) - str;
```

Comment:
No need for static cast if you use the `char_traits` thingy.

Reply:
If you prefer the `std::char_traits` solution, I can change it. It is part of the `<string>` header.

Reply:
Done.